appsec/instrumentation/http/types.go

```diff
@@ -0,0 +1,31 @@
+package http
+
+import (
+	"net/http"
+	"net/url"
+)
+
+type (
+	HandlerOperationArgs struct {
+		Method      Method
+		Host        Host
+		RequestURI  RequestURI
+		RemoteAddr  RemoteAddr
+		Headers     Header
+		QueryValues QueryValues
+		UserAgent   UserAgent
+		IsTLS       bool
+	}
+
+	HandlerOperationRes struct {
+		Status int
+	}
+
+	Method      string
+	Host        string
+	RequestURI  string
+	RemoteAddr  string
+	UserAgent   string
+	Header      http.Header
+	QueryValues url.Values
```

Comment:
So this is a long story that I'll try to keep simple 😅 

The security rules (currenly inlined in `internal/appsec/internal/protection/waf/rule.go`) describe the data they need with what we call "addresses". For example `server.request.body.raw`, `server.request.headers`, `server.request.uri.raw`, etc. So we need to map those to our Go values, through dyngo events, that are Go types.

So my idea for now was to have on Go type definition per address we support so that they can be individually listened to (via dyngo data events). An example would be:

```go
// ... walking the rule and its required addresses ...
if addr == "server.request.uri.raw" {
     op.OnData(func (_ *dyngo.Operation, httpinstr.RawRequestURI) {
         // This is the implementation of the subscription to server.request.uri.raw via dyngo
    })
}
```

And hence the `instrumentation/http` package to define in there the HTTP instrumentation types to use in HTTP instrumentation integrations.

Does it make better sense now?

Note that I plan to try to find a better data subscription API allowing to get every available data at once, instead of the current one-by-one implementation.

Reply:
And btw, a direct consequence of that is the WAF callback code that is super simple with a few lines of code, because I can write a simple "subscription loop". That made me think I was heading to the right direction. I'll try to optimize it asap though (ie. avoid the one-by-one event).

Reply:
The data event is now removed so that we make complex event chains impossible.
What was done here will still be possible and implemented in the future without the data event.